umbral/pre.py

```diff
@@ -375,45 +417,38 @@ def _decapsulate_reencrypted(receiving_privkey: UmbralPrivateKey, capsule: Capsu
     pub_key = receiving_privkey.get_pubkey().point_key
     priv_key = receiving_privkey.bn_key
 
-    precursor = capsule._attached_cfrags[0]._point_precursor
+    precursor = capsule._attached_cfrags[0].point_precursor
     dh_point = priv_key * precursor
 
-    from constant_sorrow import constants
-
     # Combination of CFrags via Shamir's Secret Sharing reconstruction
-    if len(capsule._attached_cfrags) > 1:
-        xs = [CurveBN.hash(precursor,
-                           pub_key,
-                           dh_point,
-                           bytes(constants.X_COORDINATE),
-                           cfrag._kfrag_id,
-                           params=params)
-              for cfrag in capsule._attached_cfrags]
-
-        e_summands, v_summands = list(), list()
-        for cfrag, x in zip(capsule._attached_cfrags, xs):
-            if precursor != cfrag._point_precursor:
-                raise ValueError("Attached CFrags are not pairwise consistent")
-
-            lambda_i = lambda_coeff(x, xs)
-            e_summands.append(lambda_i * cfrag._point_e1)
-            v_summands.append(lambda_i * cfrag._point_v1)
-
-        e_prime = sum(e_summands[1:], e_summands[0])
-        v_prime = sum(v_summands[1:], v_summands[0])
-    else:
-        e_prime = capsule._attached_cfrags[0]._point_e1
-        v_prime = capsule._attached_cfrags[0]._point_v1
+    xs = [hash_to_curvebn(precursor,
```

Comment:
I actually think a formal loop would be more readable here.

Reply:
✅